src/app/components/navbar/navbar.component.ts

```diff
@@ -1,15 +1,28 @@
 import { Component, OnInit } from '@angular/core';
+import {MentorService} from '../../services/mentor.service';
+import {Mentor} from '../../shared/mentor';
+import {MatDialog} from '@angular/material/dialog';
 
 @Component({
   selector: 'app-navbar',
   templateUrl: './navbar.component.html',
   styleUrls: ['./navbar.component.scss']
 })
 export class NavbarComponent implements OnInit {
-
-  constructor() { }
+  mentor: Mentor;
+  mentorName: string;
+  constructor(private mentorService: MentorService) {  }
 
   ngOnInit(): void {
+    this.getMentor();
   }
 
+  getMentor() {
+    this.mentorService.getMentor().subscribe(
```

Comment:
If you subscribe here, you have to unsubscribe somewhere, for example you can pipe this observable and have a Subject with takeUntil() that calls next() in ngOnDestroy. Or you can store mentor/mentorName as an observable and use it with `| async` and avoid thinking about unsubscribing.

Reply:
Done - used async